examples/scene/volume_depth_color_tf.py

```diff
@@ -0,0 +1,128 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+# vispy: gallery 2
+
+"""
+Volume Rendering with a Depth Color Transfer Function
+=====================================================
+
+This example demonstrates how to use a transfer function to color the volume based on the depth of
+the maximum value.
+
+Controls:
+
+* 1  - Show attenuated MIP, colored by depth of max value
+* 2  - Show attenuated MIP
+* - / = - Decrease/increase depth color scale
+* [ / ] - Decrease/increase attenuation
+
+"""
+import numpy as np
+from vispy import app, scene, io
+from vispy.color import BaseTransferFunction
+
+try:
+    import matplotlib
+    mip_colormap = "turbo"
+except ImportError:
+    matplotlib = None
+    mip_colormap = "viridis"
+
+# Read and normalize volume data
+mri_data = np.load(io.load_data_file('brain/mri.npz'))['data']
+mri_data = np.flipud(np.rollaxis(mri_data, 1)).astype(np.float32)
+mri_data = (mri_data - mri_data.min()) / np.ptp(mri_data)
+
+# Prepare canvas
+canvas = scene.SceneCanvas(keys='interactive', size=(768, 768), show=True)
+canvas.measure_fps()
+view = canvas.central_widget.add_view()
+
+
+class DepthColorTF(BaseTransferFunction):
+    """Transfer function that colors the volume based on the depth of the maximum value."""
+
+    glsl_tf = """\
+    uniform float u_depth_color_scale;
+    vec4 applyTransferFunction(vec4 color, vec3 loc, vec3 start_loc, vec3 step, float max_depth) {
+        vec3 d = loc - start_loc;
+        float depth = u_depth_color_scale * length(d) / max_depth;
+        depth = smoothstep(0.0, 1.0, depth);
+        depth = depth * (clim.y - clim.x) + clim.x;
+        vec4 hue = applyColormap(depth);
+        vec4 final_color = vec4(color.r * hue.rgb, 1.0);
+        return final_color;
+    }
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._depth_color_scale = 1.0
+
+    @property
+    def depth_color_scale(self):
+        return self._depth_color_scale
+
+    @depth_color_scale.setter
+    def depth_color_scale(self, value):
+        self._depth_color_scale = value
+
+    def get_uniforms(self):
+        return {"u_depth_color_scale": self.depth_color_scale}
```

Comment:
I love this approach, makes it intuitive to work with complex stuff!

Reply:
Thanks! Now that I look again we may want to add something like this. It doesn't matter at the moment but would if for some reason the `BaseTransferFunction` ever added a uniform.

`return super().get_uniforms() | {"u_depth_color_scale": self.depth_color_scale}`

Reply:
Hm, I still like having this but I'm realizing I can achieve what I intended by setting `vol.gamma`, since this transfer function is using `applyColormap`.